rotkehlchen/chain/evm/decoding/uniswap/v3/utils.py

```diff
@@ -0,0 +1,199 @@
+    from rotkehlchen.inquirer import Inquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+POOL_INIT_CODE_HASH: Final = '0xe34f199b19b2b4f47f68442619d555527d244f78a3297ea89325f843f87b8b54'
+POW_96: Final = 2**96
+LOG_PRICE: Final = FVal('1.0001')
+
+
+def _compute_pool_address(
+        uniswap_v3_factory_address: ChecksumEvmAddress,
+        token0_address_raw: str,
+        token1_address_raw: str,
+        fee: int,
+) -> ChecksumEvmAddress:
+    """
+    Generate the pool address from the Uniswap Factory Address, a pair of tokens
+    and the fee using CREATE2 opcode.
+
+    May raise:
+    - DeserializationError
+    """
+    token_0 = to_checksum_address(token0_address_raw)
+    token_1 = to_checksum_address(token1_address_raw)
+    # Sort the addresses
+    if int(token_0, 16) < int(token_1, 16):
+        parameters = [token_0, token_1, fee]
+    else:
+        parameters = [token_1, token_0, fee]
+
+    return generate_address_via_create2(
+        address=uniswap_v3_factory_address,
+        salt=to_hex(Web3.keccak(encode_abi(['address', 'address', 'uint24'], parameters))),
+        init_code=POOL_INIT_CODE_HASH,
+        is_init_code_hashed=True,
+    )
+
+
+def compute_sqrt_values_for_amounts(
+        tick_lower: int,
+        tick_upper: int,
+        tick: int,
+) -> tuple[FVal, FVal, FVal]:
+    """Computes the values for `sqrt`, `sqrt_a`, sqrt_b`"""
+    sqrt_a = LOG_PRICE**FVal(tick_lower / 2) * POW_96
+    sqrt_b = LOG_PRICE**FVal(tick_upper / 2) * POW_96
+    sqrt = LOG_PRICE**FVal(tick / 2) * POW_96
+    sqrt = max(min(sqrt, sqrt_b), sqrt_a)
+
+    return sqrt, sqrt_a, sqrt_b
+
+
+def calculate_amount(
+        tick_lower: int,
+        liquidity: int,
+        tick_upper: int,
+        decimals: int,
+        tick: int,
+        token_position: Literal[0, 1],
+) -> FVal:
+    """
+    Calculates the amount of a token in the Uniswap V3 LP position.
+    `token_position` can either be 0 or 1 representing the position of the token in a pair.
+    """
+    sqrt, sqrt_a, sqrt_b = compute_sqrt_values_for_amounts(
+        tick_lower=tick_lower,
+        tick_upper=tick_upper,
+        tick=tick,
+    )
+    if token_position == 0:
+        amount = (liquidity * POW_96 * (sqrt_b - sqrt) / (sqrt_b * sqrt)) / 10**decimals
+    elif token_position == 1:
+        amount = liquidity * (sqrt - sqrt_a) / POW_96 / 10**decimals
+
+    return FVal(amount)
+
+
+def get_uniswap_v3_position_price(
+        evm_inquirer: 'EvmNodeInquirer',
+        inquirer: 'Inquirer',
+        token: 'EvmToken',
+) -> Price:
+    """Get the USD price of a Uniswap V3 LP position identified by the token's collectible id."""
+    try:
+        token_id = int(token.collectible_id or '')  # also raise ValueError if id is None
+    except ValueError:
+        log.error(f'Failed to find Uniswap V3 position price for {token} due to invalid token ID.')
+        return ZERO_PRICE
```

Comment:
This seems wrong.If `collectible_id` is None it will raise `TypeError`.

```suggestion
    try:
        token_id = int(token.collectible_id)
    except TypeError:
        log.error(f'Failed to find Uniswap V3 position price for {token} due to invalid token ID.')
        return ZERO_PRICE
```

Reply:
Now using `identifier_to_collectible_id` here with an if check for `None`. The `token_id` is now being converted to `int` where it actually makes the contract call and in the unexpected event of the id containing non-integer characters I added `ValueError` to the try block there to handle it.